### PR TITLE
Pages and Elements API

### LINKED ISSUE #22: Implement Pages API
details in #7

* [x] SQL statements
* [x] Implement Routes
* [ ] Unit tests

### LINKED ISSUE #23: Implements Elements API
details in #13 

* [x] SQL statements
* [x] Implement Routes
* [ ] Unit tests

### PR DESCRIPTION
**Pull request by cadecairos**
Fixes #22 
Fixes #23 

Pages API is implemented, Elements is halfway there

Unit tests to follow.

### CODE REVIEW

File: scripts/create-tables.sql [hunk 1]
```diff
@@ -30,8 +29,42 @@ CREATE TABLE IF NOT EXISTS "projects"
...
+  CONSTRAINT elements_id_pk PRIMARY KEY (id)
+);
+
+/* Indexes */
+CREATE INDEX user_idx_id_deleted_at ON users (id, deleted_at);
+CREATE INDEX project_id_user_id_deleted_at ON projects (id, user_id, deleted_at);
+CREATE INDEX project_deleted_at_user_id on projects (deleted_at, user_id);
+CREATE INDEX project_id_deleted_at ON pages (project_id, deleted_at);
+CREATE INDEX pages_id_x_y_deleted_at ON pages (id, x, y, deleted_at);
+CREATE INDEX elements_page_is_deleted_at ON elements (id, page_id, deleted_at);
```
Comment (cadecairos) [inline scripts/create-tables.sql]: I still need to verify that all the above indexes get used for all the queries

File: services/api/lib/postgre.js [hunk 2]
```diff
@@ -80,6 +80,46 @@ exports.register = function(server, options, done) {
...
+  }, {});
+
+  server.method('elements.update', function(values, done) {
+    executeQuery(queries.elements.update, values, done);
+  }, {});
+
+  server.method('elements.remove', function(values, done) {
+    executeQuery(queries.elements.remove, values, done);
+  }, {});
+
```
Comment (cadecairos) [inline services/api/lib/postgre.js]: I could DRY this out a lot

File: scripts/create-tables.sql (hunk 1 above)
Comment (thisandagain) [inline scripts/create-tables.sql]: Getting a staging instance up w/ a monitoring lib installed (nodetime, strongloop, newrelic, etc) is a great way to test this across all resources.

File: services/api/lib/postgre.js (hunk 2 above)
Comment (thisandagain) [inline services/api/lib/postgre.js]: :+1: lots of boilerplate in here.

File: services/api/routes/authenticated.js [hunk 3]
```diff
@@ -223,14 +230,186 @@ var routes = [
...
+          user: numericSchema,
+          project: numericSchema
         }
       },
       cors: {
         methods: ['options', 'post']
       }
     }
+  }, {
+    path: '/users/{user}/projects/{project}/pages',
```
Comment (thisandagain) [inline services/api/routes/authenticated.js]: Lots of boilerplate through here.

**Comment by thisandagain** [comment]
Little concerned about the amount of boilerplate in a few places, but R+ this was an incredible effort. :+1: :cat2: :cake: :boom:

**Pull request merged**